cmd/cluster/instance_migrate.go

```diff
@@ -0,0 +1,584 @@
+
+	MigrateCmd.AddCommand(replaceVoyagerCmd)
+}
+
+func replaceVoyagerF(command *cobra.Command, args []string) error {
+	astarteNamespace, err := command.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+	operatorName, err := command.Flags().GetString("operator-name")
+	if err != nil {
+		return err
+	}
+	operatorNamespace, err := command.Flags().GetString("operator-namespace")
+	if err != nil {
+		return err
+	}
+
+	// is the migration allowed?
+	if err := ensureOperatorMinimumVersionRequirement(operatorName, operatorNamespace); err != nil {
+		return err
+	}
+
+	// ensure the Voyager CRD is present
+	_, err = kubernetesAPIExtensionsClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(),
+		"ingresses.voyager.appscode.com", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	aviList, err := kubernetesDynamicClient.Resource(aviV1Alpha1).Namespace(astarteNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	if len(aviList.Items) == 0 {
+		return errors.New("No AstarteVoyagerIngress resources are present in the cluster. Nothing to do here.")
+	}
+
+	// get the first available avi
+	aviObject := &aviList.Items[0]
+
+	// and, in case, if the ingress-name is set, override the object
+	aviResourceName, err := command.Flags().GetString("ingress-name")
+	if err != nil {
+		fmt.Println("Error parsing the ingress-name flag. The option will be neglected.", err)
+	}
+	if aviResourceName != "" {
+		for _, item := range aviList.Items {
+			if item.GetName() == aviResourceName {
+				aviObject = &item
+				break
+			}
+		}
+		if aviObject.GetName() != aviResourceName {
+			fmt.Printf("Couldn't find the %s resource. Falling back to the %s resource.\n", aviResourceName, aviObject.GetName())
+		}
+	}
+
+	fmt.Printf("You are about to migrate the AstarteVoyagerIngress named: %s. ", aviObject.GetName())
+	shouldMigrate, err := utils.AskForConfirmation("Are you sure?")
+	if err != nil {
+		return err
+	}
+	if !shouldMigrate {
+		fmt.Println("Ok, nothing left to do here.")
+		os.Exit(0)
+	}
+
+	// if required, dump the avi custom resource
+	if fn := command.Flag("out").Value.String(); fn != "" {
+		if err := dumpResourceToYAMLFile(aviObject, fn); err != nil {
+			fmt.Println("Warning: failed to save the AstarteVoyagerIngress resource to file: ", err)
+		} else {
+			fmt.Printf("%s resource successfully written to %s\n", aviObject.GetName(), fn)
+		}
+	}
+
+	adiName, err := utils.PromptChoice("Choose the new AstarteDefaultIngress name:", "adi", false, false)
+	if err != nil {
+		return err
+	}
+
+	// We are not checking if the secrets are present in the cluster: if they are not, the validation webhook will return an error
+	apiSecretName, err := utils.PromptChoice("Insert the name of the secret containing the TLS certificates and keys to connect to the Astarte API and Dashboard:", "", false, false)
```

Comment:
A user may arrive here after having just created the `astarte-tls-secret` using the [Handling Astarte certificates](https://docs.astarte-platform.org/1.0/050-handling_certificates.html) guide. We should display the `apiSecretName` as default value, as we do in the next step for the broker

Reply:
We have no way of knowing in advance which is the secret to be used and therefore we can not suggest a reasonable default.